### DISCUSSION TITLE
How to persist none primitive type in Nuxt 3?

### DISCUSSION THREAD

**Discussion by pheng0077** [iendeavor/pinia-plugin-persistedstate-2#29] [Q&A] [answered]
I'm trying to persist with the object, but it's not working. Are there any solutions to persist objects?

**Comment by soc221b** [comment, marked as answer]
Use [`serialize` and `deserialize`](https://github.com/iendeavor/pinia-plugin-persistedstate-2#serialization-and-deserialization), for example:
https://github.com/iendeavor/pinia-plugin-persistedstate-2/blob/678ef5b5218daadc01f55709c4499607fdfc0b0e/examples/xstate-example/src/stores/water.ts#L25-L34
